src/druidry/data_source.py

```diff
@@ -211,6 +223,23 @@ def contains_filter(f):
         return filter_.negate()
 
 
+def regex_contain_filter(f):
+    if f['left']['type'] != 'field' or f['right']['type'] != 'value':
+        raise ValueError('Druid does not support dynamic patterns.')
+    pattern = "|".join([('.*{}.*').format(value) for value in f['right']['value']])
+    if f['type'] == 'not contains':
+        return RegexFilter(
+        dimension=f['left']['field'], pattern=pattern).negate()
+    return RegexFilter(
+        dimension=f['left']['field'], pattern=pattern)
+
+def in_filter(f):
+    if f['left']['type'] != 'field' or f['right']['type'] != 'value':
+        raise ValueError('Druid does not support dynamic containment checks.')
```

Comment:
In L#158-174 we check both left and right for 'field', but we raise an error here if 'field' is not f[left][type]. Is this intentional? Is there a difference in use between the filters?

Reply:
I have used it after checking all other filters, let me know if you are sure we dun need it

Reply:
I think it's intentional. There are three classes of comparisons seen in this file:
1. field to field
2. field to constant
3. constant to constant

Aside from 3, which isn't really useful, we implement the others as needed in here depending on where that operator's being used. Since for this we're only using them in AE, where the comparisons are only field to constant, I think only allowing that case here is fine. The exception message text might not be 100% correct... think we can leave it alone though.